x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -70,6 +90,159 @@ export function LayerContextMenu(props: VisualizationLayerWidgetProps<State>) {
+                        <p>{description}</p>
+                      </EuiText>
+                    </>
+                  ),
+                  inputDisplay: title,
+                };
+              })}
+              valueOfSelected={props.state?.fittingFunction || 'None'}
+              onChange={(value) => props.setState({ ...props.state, fittingFunction: value })}
+              itemLayoutAlign="top"
+              hasDividers
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            display="columnCompressed"
+            label={i18n.translate('xpack.lens.xyChart.legendVisibilityLabel', {
+              defaultMessage: 'Legend display',
+            })}
+          >
+            <EuiButtonGroup
+              legend={i18n.translate('xpack.lens.xyChart.legendVisibilityLabel', {
+                defaultMessage: 'Legend display',
+              })}
+              name="legendDisplay"
+              buttonSize="compressed"
+              className="eui-displayInlineBlock"
+              options={[
+                {
+                  id: `${idPrefix}auto`,
+                  label: i18n.translate('xpack.lens.xyChart.legendVisibility.auto', {
+                    defaultMessage: 'auto',
+                  }),
+                },
+                {
+                  id: `${idPrefix}show`,
+                  label: i18n.translate('xpack.lens.xyChart.legendVisibility.show', {
+                    defaultMessage: 'show',
+                  }),
+                },
+                {
+                  id: `${idPrefix}hide`,
+                  label: i18n.translate('xpack.lens.xyChart.legendVisibility.hide', {
+                    defaultMessage: 'hide',
+                  }),
+                },
+              ]}
+              idSelected={`${idPrefix}${legendMode}`}
+              onChange={(id) => {
+                const newMode = id.replace(idPrefix, '') as 'auto' | 'hide' | 'show';
+                if (newMode === 'auto') {
+                  props.setState({
+                    ...props.state,
+                    legend: { ...props.state.legend, isVisible: true, showSingleMetric: false },
+                  });
+                } else if (newMode === 'show') {
+                  props.setState({
+                    ...props.state,
+                    legend: { ...props.state.legend, isVisible: true, showSingleMetric: true },
+                  });
+                } else if (newMode === 'hide') {
+                  props.setState({
+                    ...props.state,
+                    legend: { ...props.state.legend, isVisible: false, showSingleMetric: false },
+                  });
+                }
+              }}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            display="columnCompressed"
+            label={i18n.translate('xpack.lens.xyChart.legendPositionLabel', {
+              defaultMessage: 'Legend position',
+            })}
+          >
+            <EuiSelect
```

Comment:
Always add the `compressed` prop to inputs inside popovers and especially inside `columnCompressed` form rows.